src/ray/core_worker/task_event_buffer.h

```diff
@@ -57,6 +58,14 @@ class TaskEvent {
   /// \param[out] rpc_task_events The rpc task event to be filled.
   virtual void ToRpcTaskEvents(rpc::TaskEvents *rpc_task_events) = 0;
 
+  /// Convert itself a rpc::ExportTaskEventData
+  ///
+  /// NOTE: this method will modify internal states by moving fields to the
```

Comment:
Let's remove the comment and make sure data is not modified inside this function